File: backend/src/main/java/com/interview/model/Book.java
package com.interview.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "book")
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;
    private String isbn;
    private Integer publicationYear;
    private String publisher;
    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinTable(
            name = "book_author",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "author_id")
    )
    private Set<Author> authors;
}

File: frontend/src/types/book.ts
import Author from "./author";

export default interface Book {
    id: number | null;
    name: string | null;
    publicationYear: number | null;
    author: Author | null;
}
  
export interface PageOfBooks {
    currentPage: number;
    totalPage: number;
    totalItems: number;
    books : Book[];
}